### PR TITLE
feat: add unfinished_candle method

### PR DESCRIPTION
**Pull request by ivaaaan**
Another problem I have faced, is that there is no way to get the latest candle from the aggregator.

Input data:

```
timestamp,price,size
1612137602564000,27334.25,0.00289
1612137664747000,27298.5,0.00909044
```

`TimeRule` and `AlignedTimeRule` will trigger on a first trade, and will return the candle, but there is no way to get the second candle. 

I propose to add a new state for candle - `finalized`, and method `close` to aggregator. This way, user can easily retrieve the latest candle from the aggregator. 

Let me know if there are other ways to achieve this.

### CODE REVIEW

File: trade_aggregation_derive/src/lib.rs
Comment (ivaaaan) [inline trade_aggregation_derive/src/lib.rs:1]: I'm not familiar with Rust, so this is just my try to add a default implementation for trait methods. Of course it doesn't work, so please let me know how to do it correctly :)

File: src/aggregator.rs [hunk 1]
```diff
@@ -13,6 +13,12 @@ pub trait Aggregator<Candle, T: TakerTrade> {
     /// Some output only when a new candle has been created,
     /// otherwise it returns None
     fn update(&mut self, trade: &T) -> Option<Candle>;
+
+    /// Close current active candle and return it
+    /// 
+    /// # Returns:
+    /// Last active candle, or None if candle was already finalized
+    fn close(&mut self) -> Option<Candle>;
```
Comment (MathisWellmann) [inline src/aggregator.rs]:
```suggestion

    /// Get a reference to an unfinished `Candle`.
    /// Accessing a `Candle` using this method does not guarantee that the `AggregationRule` is respected.
    /// It is generally advised to call `update` instead and use the resulting `Candle` if its `Some`.
    fn unfinished_candle(&self) -> &Candle;
```

**Review by MathisWellmann — COMMENTED** [review]
I'd prefer simply adding a `unfinished_candle` getter method which provides even more flexibility compared to simply closing an `Aggregator`.
Its a good feature proposal overall, enhancing the flexibility.
We should be careful to ensure that its clear to a library user that accessing a `Candle` through a method like this is not recommended as it does not respect the `AggregationRule`. But for more advanced use cases this is useful

File: src/aggregator.rs (hunk 1 above)
Comment (ivaaaan) [reply src/aggregator.rs]: I agree, that's what I ended up doing - implemented my own aggregator with method like this. Will update the PR

**Comment by ivaaaan** [comment]
@MathisWellmann updated :)

**Review by MathisWellmann — APPROVED** [review]

**Pull request merged**